src/ui/control/geolocate_control.js

```diff
@@ -476,6 +502,28 @@ class GeolocateControl extends Evented {
 
 export default GeolocateControl;
 
+function getDistance(latlng1, latlng2) {
+    // Uses spherical law of cosines approximation.
+    const R = 6371000;
+
+    const rad = Math.PI / 180,
+        lat1 = latlng1.lat * rad,
+        lat2 = latlng2.lat * rad,
+        a = Math.sin(lat1) * Math.sin(lat2) +
+          Math.cos(lat1) * Math.cos(lat2) * Math.cos((latlng2.lng - latlng1.lng) * rad);
+
+    const maxMeters = R * Math.acos(Math.min(a, 1));
+    return maxMeters;
+}
+
+function updateCircleRadius(map, circle, accuracy) {
+    const y = map._container.clientHeight / 2;
+    const metersPerPixel = getDistance(map.unproject([0, y]), map.unproject([1, y]));
+    const circleDiameter = Math.ceil(2.0 * accuracy / metersPerPixel);
+    circle.style.width = `${circleDiameter}px`;
+    circle.style.height = `${circleDiameter}px`;
+}
+
```

Comment:
Nit: for consistency, since we're passing 3 arguments here including the map instance, I think this should live as an `_updateCircle()` method on the control class.